### ISSUE TITLE
NVDA 2021.1 and newer. A problem with some items inside some groupings

### ISSUE THREAD

**Issue by HanaDadfar** [nvaccess/nvda#16252]
today, I was remembering a case where a long time ago, NVDA Used to read the title of some edit boxes inside groupings, but it does not do that anymore. For example, in nvda 2020.4 and older, when we were creating a portable copy of nvda, for the portable directory edit box, we would hear Portable directory edit blank or portable directory edit selected C:\users\blaablaa\blaablaa, but in nvda 2021.1 and newer, we just hear edit blank, or edit selected blaablaa. So the title doesn't announced. or when we are exploring the synthesizer grouping, we have a change button and a read only multy line edit box. in nvda 2020.4 and older, when we were focused on this box, nvda would read it like this. Synthesizer edit  read only  multi line alt+s  eSpeak NG but in nvda 2021.1 and newer, it is announced like this edit  read only  multi line  eSpeak NG and it is like this until now, and haven't been fixed. I don't know if this happens anywhere else, but i only figgered out the box for Synthesizer, the box for Braille display, and the box for selecting portable directory.
### Steps to reproduce:
navigate the Synthesizer grouping and focus on its read only edit box, and also do the same for the portable directory edit box, and the read only edit box for Braille display. Compare all of these with nvda 2020.4, and 2021.1.
### Actual behavior:
nvda doesn't announce the title of these boxes, and their shortcuts if they have any.
### Expected behavior:
nvda should announce the title of these boxes, and their shortcuts if they have any.
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2024.1 beta 10 and older down to 2021.1 has this problem
#### Windows version:
Windows 10 enterprise 22H2 (AMD64) build 19045.4046
#### Name and version of other software in use when reproducing the issue:
non
#### Other information about your system:
non
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2020.4 and older works perfectly fine
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by Emil-18** [comment]
@HanaDadfar I think this is because the information is reported in the group box that the edit field is inside of. If you tab from the categories list and into for example the synthesizer edit box, you should here something like this.
"Speech  property page, Synthesizer  grouping, edit  read only  IBMTTS"

**Comment by Adriani90** [comment]
Yes this is expected and is so by design, it announces the title of the edit box only when you enter the grouping with tab or shift+tab. The edit box is grouped with the button next to it, they both have the same grouping title.
The shortcuts alt+s part is not available anymore for groupings, but only for interactive elements such as buttons, lists, combo boxes, checkboxes and edit fields. alt+s is now used for "use spelling functionality".
I am closing as invalid

**Comment by lukaszgo1** [comment]
While this is certainly low priority, I don't think this issue should be closed. In NVDA 2020.4 it was possible to report the label of the edit field on demand by pressing `NVDA+TAB` but in 2021.1 and later it does not work anymore. The speech reporting for the edit field is also inconsistent depending if you are entering the grouping or just shift tabbing into the edit box from the 'change' button. I'm not sure if this was a WX change or something in our code has caused this.

**Comment by Adriani90** [comment]
@lukaszgo1 I guess the edit field inherits its label from the grouping title, so announcing it when landing on the edit field with shift+tab from the change button would on the other hand cause a double announcement of its label when tabbing to the edit field i.e. from the categories list.
I am not sure whether it is possible to define different reportings between shift+tab and tab in wx python at all.

I am reopening for further investigation.

**Comment by HanaDadfar** [comment]
@Adriani90 hi.
i was considering your point that may be this had happned because the announcement for the grouping is enough, but then i had some other reasons/faqts that has made me to open this issue.
1: the groupebox can completely have a difrent label than that edit box for example.
2: a groupebox can holde unlimited amount of edit boxes, checkboxes, combo boxes, spin boxes, sliders, which will have a difrent names and labels. Consider we have these 2 change button and the edit read only box in this grouping, and consider we don't know what's the label of this groupebox, and we are shift tabbing from the change button to the edit box. Then, by hearing just edit read only blaablaa, we can only guess what is there, or this is even related to the change button besides it (also considering the faqt that this case was the simple one and we could have a more complicated case for this, for now i'v only figgered out about the portable directory and this).

**Comment by Adriani90** [comment]
@HanaDadfar note that NVDA has always this issue, there is not a aria-labelledby equivalent for groupings in WX dialogs. So the only way to handle this is by
1. The software developers label properly the edit fields (maybe @seanbudd can look into this and put a label like "address" on the edit field), or
2. NVDA provides a key stroke to report the grouping label on demand which was possible with nvda+tab as described by @lukaszgo1 if I understand it correctly. As of now it works with object navigation, you can press nvda+shift+up arrow (laptop keyboard layout) and you will hear the label for the grouping.
